src/download-plugin-manager.tsx

```diff
@@ -1,32 +1,39 @@
+import {h} from 'preact';
 import {KalturaAttachmentAsset} from './providers';
 
-const {Icon} = KalturaPlayer.ui.components;
+import {core, ui} from '@playkit-js/kaltura-player-js';
+const {Icon} = ui.components;
+const {FakeEvent} = core;
 
 import {Download} from './download';
 import {DOWNLOAD, ERROR} from './icons';
 import {DownloadService} from './services';
 import {DownloadMetadata} from './types';
 import {DownloadEvent} from './event';
 
-class DownloadPluginManager extends KalturaPlayer.core.FakeEventTarget {
+class DownloadPluginManager extends core.FakeEventTarget {
   private _showOverlay = false;
   private downloadService: any;
   private downloadMetadata: DownloadMetadata = null;
   private playOnClose = false;
 
   constructor(public downloadPlugin: Download) {
     super();
+    // @ts-expect-error - Property 'player' is protected and only accessible within class 'BasePlugin' and its subclasses.
```

Comment:
you could also pass player, config and logger to the constructor. what i don't like about ts-ignore (and i'm not sure if ts-expect-error behaves the same way) is that it will ignore ANY error on that line, leading to possible bugs in the future